parcels/_index_search.py

```diff
@@ -218,17 +218,14 @@ def _search_indices_rectilinear(
         _raise_field_sampling_error(z, y, x)
 
     if particle:
-        particle.xi[field.igrid] = xi
-        particle.yi[field.igrid] = yi
-        particle.zi[field.igrid] = zi
+        particle.ei[field.igrid] = field.ravel_index(xi, yi, zi)
 
     return (zeta, eta, xsi, zi, yi, xi)
 
 
 def _search_indices_curvilinear(field: Field, time, z, y, x, ti=-1, particle=None, search2D=False):
     if particle:
-        xi = particle.xi[field.igrid]
-        yi = particle.yi[field.igrid]
+        zi, yi, xi = field.unravel_index(particle.ei[field.igrid])
```

Comment:
Would it be possible to get the `field.igrid` here from the `field` that calls the function? so instead implement
```suggestion
        zi, yi, xi = field.unravel_index(particle.ei)
```

Reply:
It's possible, but I'm not sure how this kernel in `test/test_particlefile.py` would work.. 
https://github.com/OceanParcels/Parcels/pull/1878/files#diff-14e032bcf23d770966c77a78ba9016961b70b8c5a1928e7022717708031d0268R276

Reply:
Perfect. Got this squared a way in https://github.com/OceanParcels/Parcels/pull/1878/commits/42770dc3d00398e663a0cc378429c50060cb1423
Had to adjust the `test_multi_structured_grids` test assertions as well, but this should be good to go now.